Feat/add should upload event support

BrandonStalnaker opened the pull request:
# Summary

Updates the Core SDKs and adds new bypass upload functionality.

willpassidomo commented on Assets/mParticle/MParticleAndroid.cs:
```diff
@@ -111,6 +111,10 @@ public void LogEvent(CommerceEvent commerceEvent)
 			{
 				builder.Call<AndroidJavaObject>("nonInteraction", (bool)commerceEvent.NonInteractive);
 			}
+			if (commerceEvent.ShouldUploadEvent.HasValue)
```
oh cool...didn't know they added the `HasValue` method but that is pretty nifty

willpassidomo commented:
So, the Android implentation _looks_ good, but it doesn't compile. Seems like when we updated `Assets/Plugins/Android/mparticle-core.jar` the new one is actually our Gradle Plugin (one we built for dataplanning), not the core SDK

proper file to use would be the `aar` file in this directory: https://repo1.maven.org/maven2/com/mparticle/android-core/5.25.0/
I can't remember if Unity requires a `.jar` or if the `.aar` is ok, but if the former then you can open the `.aar` as you would a `.zip` file and use the `classes.jar` file inside

also, @BrandonStalnaker very my bad, I just looked back at the slack message you sent me and realized I skimmed it and didn't look at the entire link, I should have taken a little extra time and noticed that was the wrong one in the first place 😬

BrandonStalnaker commented:
@willpassidomo I'm pretty sure it needs the .jar so I extracted the 'classes.jar' and renamed it to 'mparticle-core.jar'. Please run this to be sure cause because I have the sneaking suspicion that when I'm testing this its not using the android code like I think I should be.

willpassidomo reviewed (approved):
This looks good, I can get it to build in Unity for Android, but haven't wrangled testing runtime behavior...honestly, I'm probably being a little extra on this one 😆

einsteinx2 reviewed (approved):
Will is confirming the iOS build since I don't have a Unity environment, but iOS code LGTM.
